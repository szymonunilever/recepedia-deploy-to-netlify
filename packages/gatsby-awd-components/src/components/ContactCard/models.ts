import {
  titleLevel,
  UnileverLibraryComponent,
} from '../../models/globalModels';

export interface ContactCardProps extends UnileverLibraryComponent<AppContent.ContactCard> {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  Icon?: any;
  titleLevel: titleLevel;
}
