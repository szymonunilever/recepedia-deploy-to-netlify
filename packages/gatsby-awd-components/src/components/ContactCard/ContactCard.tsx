import cx from 'classnames';
import React from 'react';
import Button from '../Button';
import { TagName, Text } from '../Text';
import theme from './ContactCard.module.scss';
import { ContactCardProps } from './models';
import { navigate } from 'gatsby';
import getComponentDataAttrs from '../../utils/getComponentDataAttrs';

export const ContactCard = ({
  className,
  content,
  content: { title, texts, type, cta },
  Icon,
  titleLevel = 2,
}: ContactCardProps) => {
  const classWrapper = cx(theme.contactCard, className);

  const Title = (
    <Text
      // @ts-ignore
      tag={TagName[`h${titleLevel}`]}
      text={title}
      className={cx(theme.contactCard__title, 'contact-card__title')}
    />
  );

  const withType = (
    type: string,
    view: JSX.Element[] | false,
    content: string[] = texts
  ) => {
    if (view) {
      switch (type) {
        case 'phone':
          return <a href={`tel:${content[0]}`}>{view}</a>;
        case 'address':
          return <address>{view}</address>;
        case 'email':
          return <a href={`mailto:${content[0]}`}>{view}</a>;
        default:
          return view;
      }
    }
  };

  const view =
    texts &&
    texts.length > 0 &&
    texts.map((text, key) => (
      <Text
        key={key}
        tag={TagName.p}
        text={text}
        className={cx(theme.contactCard__text, 'contact-card__text')}
      />
    ));

  const ctaClickHandler = () => {
    if (cta && cta.linkTo) navigate(cta.linkTo);
  };

  return (
    <div {...getComponentDataAttrs('contactCard', content)} className={classWrapper}>
      <div className={cx(theme.contactCard__top, 'contact-card__top')}>
        {Icon && (
          <div className={cx(theme.contactCard__icon, 'contact-card__icon')}>
            <Icon />
          </div>
        )}
        {Title}
      </div>
      <div className={cx(theme.contactCard__content, 'contact-card__content')}>
        {withType(type, view)}
        {cta && (
          <Button
            className={cx(theme.contactCard__button, 'contact-card__button')}
            content={cta}
            onClick={ctaClickHandler}
          />
        )}
      </div>
    </div>
  );
};

export default ContactCard;
