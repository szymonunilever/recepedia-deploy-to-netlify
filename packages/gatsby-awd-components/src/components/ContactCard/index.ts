export { default, ContactCard } from './ContactCard';
export * from './models';
