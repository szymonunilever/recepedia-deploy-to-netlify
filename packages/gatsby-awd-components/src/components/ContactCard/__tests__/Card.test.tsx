import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { ContactCard } from '../ContactCard';
import toJson from 'enzyme-to-json';
import cardsData from 'src/mocks/cards.json';

describe('<ContactCard />', () => {
  let wrapper: ReactWrapper;
  const cards: any = cardsData;

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    wrapper = mount(<ContactCard {...cards[0]} Icon="svg" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should show address card', () => {
    wrapper = mount(<ContactCard {...cards[1]} Icon="svg" />);
    expect(wrapper.find('address')).toHaveLength(1);
  });

  it('should show text card', () => {
    wrapper = mount(<ContactCard {...cards[2]} Icon="svg" />);
    expect(wrapper.find('.card__text')).toHaveLength(2);
  });
});
