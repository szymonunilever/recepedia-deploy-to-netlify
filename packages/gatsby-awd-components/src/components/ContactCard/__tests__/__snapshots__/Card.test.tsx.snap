// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactCard /> matches the snapshot 1`] = `
<Component
  Icon="svg"
  content={
    Object {
      "cta": Object {
        "label": "Contact Form",
        "linkTo": "/contact",
      },
      "texts": Array [
        "+380633334455",
      ],
      "title": "Phone",
      "type": "phone",
    }
  }
  titleLevel={2}
>
  <div
    className="contactCard"
    data-componentname="contactCard"
    data-componentview="contactCard"
  >
    <div
      className="contactCard__top contact-card__top"
    >
      <div
        className="contactCard__icon contact-card__icon"
      >
        <svg />
      </div>
      <Text
        className="contactCard__title contact-card__title"
        tag={1}
        text="Phone"
      >
        <h2
          className="contactCard__title contact-card__title"
        >
          Phone
        </h2>
      </Text>
    </div>
    <div
      className="contactCard__content contact-card__content"
    >
      <a
        href="tel:+380633334455"
      >
        <Text
          className="contactCard__text contact-card__text"
          key="0"
          tag={7}
          text="+380633334455"
        >
          <p
            className="contactCard__text contact-card__text"
          >
            +380633334455
          </p>
        </Text>
      </a>
      <Component
        className="contactCard__button contact-card__button"
        content={
          Object {
            "label": "Contact Form",
            "linkTo": "/contact",
          }
        }
        onClick={[Function]}
      >
        <button
          className="button button contactCard__button contact-card__button"
          data-componentname="button"
          data-componentview="button"
          disabled={false}
          hidden={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="button__label"
          >
            Contact Form
          </span>
        </button>
      </Component>
    </div>
  </div>
</Component>
`;
